include/arch/arc/arch.h

```diff
@@ -38,7 +38,107 @@ extern "C" {
 #include <arch/arc/v2/addr_types.h>
 #endif
 
+#if defined(CONFIG_MPU_STACK_GUARD)
+#if defined(CONFIG_ARC_CORE_MPU)
+#if CONFIG_ARC_MPU_VER == 2
+/* The minimum MPU region of MPU v2 is 2048 bytes */
+#define STACK_ALIGN  2048
+#elif CONFIG_ARC_MPU_VER == 3
+#define STACK_ALIGN 32
+#endif
+#else /* CONFIG_ARC_CORE_MPU */
+#error "Unsupported STACK_ALIGN"
+#endif
+#else  /* CONFIG_MPU_STACK_GUARD */
 #define STACK_ALIGN  4
+#endif
+
+/**
+ * @brief Declare a toplevel thread stack memory region
```

Comment:
these APIS are already documented in kernel.h, please don't copy-paste the documentation here